how-to-use-azureml/machine-learning-pipelines/intro-to-pipelines/aml-pipelines-schedule-with-pipeline-parameter-built-in-functions.ipynb

```diff
@@ -0,0 +1,296 @@
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "recurring_schedule.update(name=\"NewRecurringSchedule\",\n",
+    "                description=\"Based on time, pipeline param expression\",\n",
+    "                pipeline_parameters=update_pipeline_parameters,\n",
+    "                recurrence=update_recurrence)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Create a schedule for pipeline endpoint"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "- Publish or get a pipeline endpoint"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from azureml.pipeline.core import PipelineEndpoint\n",
+    "pipeline_endpoint_name = \"My_PipelineEndpoint\"\n",
+    "cur_pipeline = all_pub_pipelines[0]\n",
+    "try:\n",
+    "    pipeline_endpoint = PipelineEndpoint.publish(workspace=ws,\n",
+    "                                                name=pipeline_endpoint_name,\n",
+    "                                                pipeline=cur_pipeline,\n",
+    "                                                description=\"publish pipeline endpoint\")\n",
+    "except Exception as e:\n",
+    "    if \"already exists with the given name\" in e.args[0]:\n",
+    "        pipeline_endpoint = PipelineEndpoint.get(workspace=ws, name=pipeline_endpoint_name)\n",
+    "        print(\"pipeline_endpoint already exists\")\n",
+    "    else:\n",
+    "        raise e"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "- Schedule a pipeline endpoint"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "pipeline_endpoint_id = pipeline_endpoint.id\n",
+    "pipeline_endpoint_schedule = Schedule.create_for_pipeline_endpoint(ws,\n",
+    "                                    name=\"MyPipelineEndpointSchedule\",\n",
+    "                                     pipeline_endpoint_id=pipeline_endpoint_id,\n",
+    "                                     experiment_name=\"endpoint-pipeline-parameter-sample\",\n",
+    "                                     description=\"create for pipeline endpoint with pipeline parameters\",      \n",
+    "                                     pipeline_parameters=pipeline_parameters,\n",
+    "                                     recurrence=recurrence\n",
+    "                                     ) "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Disable schedule"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "recurring_schedule.disable()\n",
+    "pipeline_endpoint_schedule.disable()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Note\n",
```

Comment:
Let's make the part as "Questions and Answers"

Q1: Why are my input pipeline parameter expressions not resolved?
Q2: What should I do when I meet error "Invalid pipeline parameter expression"?
Q3: Why do new triggered runs still use previous pipeline parameter expressions  although I have republished with new expressions?